intrinsic_and_extrinsic_optimise_real_data/src/camera_calibration_real_data.cpp

```diff
@@ -0,0 +1,422 @@
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/calib3d/calib3d.hpp>
+#include "opencv2/highgui/highgui.hpp"
+#include <vector>
+#include "ceres/ceres.h"
+#include "ceres/rotation.h"
+
+// Definitions for the dimensions of the calibration board
+static const int Number_of_internal_corners_x = 7;
+static const int Number_of_internal_corners_y = 6;
+static const double Block_dimension = 0.02; // This is the side of a block in m.
```

Comment:
You likely already have GFLAGS installed since it's bundled with ceres.  Consider switching these constants to be gflag arguments.